lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex

```diff
@@ -54,18 +77,44 @@ defmodule Archethic.Networking.IPLookup.NATDiscovery.MiniUPNP do
       # Remote port to open
       "#{port}",
       # Protocol
-      "tcp",
+      @protocol,
       # Lifetime
       "0"
     ]
+  end
 
-    case System.cmd(@upnpc, opts) do
-      {_, 0} ->
-        :ok
+  @spec parse_reason(String.t()) ::
+          {:error, :conflict_in_mapping_entry | :unknown}
+  defp parse_reason(reason) do
+    # upon more condtions , refactor with cond do end
+    if Regex.scan(~r/ConflictInMappingEntry/, reason, capture: :all) != [] do
+      Logger.warning("Port is employed to another host.")
+      {:error, :conflict_in_mapping_entry}
```

Comment:
The error tuple is more used to help to branch between success and failure.
Here you just want to format error's reason.
I think you could return simplify the error's reason, so you could easily use this function in a pipeline.